### PR TITLE
Add `prod_sequence` to avoid creating `vector`

### PR DESCRIPTION
**Pull request by okuta**

### CODE REVIEW

**Comment by okuta** [comment]
Jenkins, test this please

**Comment by okuta** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
@okuta For performance improvements, it's better to indicate the actual measurement.

In my environment, with a tuple with 5 elements:
`prod`: 0.11 us
`prod_sequence`: 0.21 us

File: tests/cupy_tests/core_tests/test_internal.py [hunk 1]
```diff
@@ -18,16 +18,16 @@ def test_two(self):
         self.assertEqual(internal.prod([2, 3]), 6)
 
 
-class TestProdSsizeT(unittest.TestCase):
```
Comment (niboshi) [inline tests/cupy_tests/core_tests/test_internal.py]: Please keep old tests.

Comment (okuta) [reply tests/cupy_tests/core_tests/test_internal.py]: This core is same as L9-18.

Comment (niboshi) [reply tests/cupy_tests/core_tests/test_internal.py]: Oh, I didn't notice... 🤔

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
LGTM

**Pull request merged**